### PR TITLE
Adds support for sysconf

### PR DESCRIPTION
**Pull request by feliperodri**
### Description of changes:

Adds support for `sysconf`.

### Resolved issues:

Unblock customer.

### Testing:

* How is this change tested? Regression test.

* Is this a refactor change? No.

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [ ] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

**Review by celinval — APPROVED** [review]
Thanks

File: tests/kani/Intrinsics/Sysconf/main.rs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+//
+//! Check support for `sysconf`.
+
+use libc;
+
+#[kani::proof]
+fn main() {
+    let page_size = unsafe { libc::sysconf(libc::_SC_PAGESIZE) } as usize;
```
Comment (celinval) [inline tests/kani/Intrinsics/Sysconf/main.rs:10]: Question, how is sysconf modeled? Is there any check that we can ensure we get the correct result?

File: tests/kani/Intrinsics/Sysconf/main.rs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// Copyright Kani Contributors
```
Comment (celinval) [inline tests/kani/Intrinsics/Sysconf/main.rs:1]: Can you please move this test to:
`tests/kani/LibC/sysconf.rs`

**Review by celinval — COMMENTED** [review]
Just a few comments on the test. :)

File: tests/kani/Intrinsics/Sysconf/main.rs (hunk 1 above)
Comment (adpaco-aws) [reply tests/kani/Intrinsics/Sysconf/main.rs:10]: I'm also curious about this... Do we get a nondet. `usize` value?

Comment (feliperodri) [reply tests/kani/Intrinsics/Sysconf/main.rs:10]: This pull request depends on https://github.com/diffblue/cbmc/pull/7780. I'm still working on the model, which so far is just an over-approximation of the [spec](https://man7.org/linux/man-pages/man3/sysconf.3.html).

**Comment by feliperodri** [comment]
https://github.com/model-checking/kani/pull/2614 will unblock this PR (we will have access to the new model for `sysconf` in CBMC).

File: tests/kani/Intrinsics/Sysconf/main.rs (hunk 2 above)
Comment (feliperodri) [reply tests/kani/Intrinsics/Sysconf/main.rs]: Implemented as suggested.

**Pull request merged**